docs/sections/workflows/writing_basics.md

```diff
@@ -175,103 +154,445 @@ See the {ref}`corresponding topics section for more details <topics:processes:us
 Since the result of the addition is only available once the calculation job is finished, the `submit()` method returns the {class}`~aiida.orm.nodes.process.calculation.calcjob.CalcJobNode` of the *future* `ArithmeticAddCalculation` process.
-To tell the work chain to wait for this process to finish before continuing the workflow, we return the `ToContext` class, where we have passed a dictionary to specify that the future calculation job node should be assigned to the `'addition'` context key.
+To tell the work chain to wait for this process to finish before continuing the workflow, we return the `ToContext` class, where we have passed a dictionary to specify that the future calculation job node `calc_job_node` should be assigned to the `'addition'` context key.
 
-:::{warning}
+We added a new step in the outline:
 
-Never use the global `submit()` function to submit calculations to the daemon within a {class}`~aiida.engine.processes.workchains.workchain.WorkChain`.
-Doing so will raise an exception during runtime.
-See the {ref}`topics section on work chains<topics:workflows:usage:workchains:submitting_sub_processes>` for more details.
+```{literalinclude} include/code/new_scripts/multiply_add_workchain_calcjob.py
+:language: python
+:pyobject: MultiplyAddWorkChain.gather_results
+:dedent: 4
+```
 
-:::
+This final step is to gather and pass the results to the outputs of the work chain using the `self.out()` method.
 
-:::{note}
+Now, it is time to run our work chain.
+In order that the AiiDA daemon knows where to find the work chain, we need to add its directory to the `PYTHONPATH`.
+Navigate until that directory in the terminal and execute (adjusting the path for your python environment):
+
+```{code-block} console
+$ echo "export PYTHONPATH=\$PYTHONPATH:$PWD" >> /home/max/.virtualenvs/aiida/bin/activate
+$ verdi daemon restart --reset
+```
 
-Instead of passing a dictionary, you can also initialize a `ToContext` instance by passing the future process as a keyword argument, e.g. `ToContext(addition=calcjob_node)`.
-More information on the `ToContext` class can be found in {ref}`the topics section on submitting sub processes<topics:workflows:usage:workchains:submitting_sub_processes>`.
+Setup the computer and code if needed:
+
+```{code-block} console
+$ verdi computer setup --config computer.yml
+$ verdi computer configure local localhost
+$ verdi code setup --config code_add.yml
+```
+In the `verdi shell`, run:
+```{code-block} console
+In [1]: from aiida.engine import submit
+In [2]: from multiply_add_workchain_calcjob import MultiplyAddWorkChain
+In [3]: add_code = load_code(label='add@localhost')
+In [4]: inputs = {'x': Int(1), 'y': Int(2), 'z': Int(3), 'code': add_code}
+In [5]: workchain_node = submit(MultiplyAddWorkChain, **inputs)
+```
+
+Now, go and check the last processes, the status of the `MultiplyAddWorkChain`, and show some details about the inputs and outputs.
+
+
+#### Checks and validation
```

Comment:
As we've already discussed, this should be moved to a separate module. Note that another advantage of working with smaller pieces is that by now I've been reviewing too much material to still focus. Smaller PR's make for better reviews!

Reply:
I suggest I move and adapt this, and then you review :)